### PR TITLE
feat(snyk): add snyk linter

### PR DESCRIPTION
**Pull request by pbnj**
## TODO 

- [x] add tests
- [x] list tool in readme

### CODE REVIEW

File: lua/lint/linters/snyk_iac.lua [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      return diagnostics
+    end
+    for _, result in ipairs(decoded and decoded.infrastructureAsCodeIssues or {}) do
+      local err = {
+        source = "snyk",
+        message = string.format("%s - %s - %s", result.title, result.issue, result.impact),
+        lnum = result.lineNumber - 1,
+        end_lnum = result.lineNumber - 1,
+        col = result.lineNumber,
+        end_col = result.lineNumber,
```
Comment (mfussenegger) [inline lua/lint/linters/snyk_iac.lua:28]: Using the lineNumber as columns seems wrong?

Comment (pbnj) [reply lua/lint/linters/snyk_iac.lua:28]: I copied it from [`tfsec` linter](https://github.com/mfussenegger/nvim-lint/blob/6d596b87862909370f4d1e6535cc9fad80c89fe6/lua/lint/linters/tfsec.lua#L24-L27), but I agree.

If the linter does not provide a column, should I leave `col` and `end_col` out? Or should I set them to `1`?

Comment (pbnj) [reply lua/lint/linters/snyk_iac.lua]: Removed `col` and `end_col`.

Comment (mfussenegger) [reply lua/lint/linters/snyk_iac.lua]: `col` is required according to `:help diagnostic-structure`, so if it isn't provided by the linter it should probably be set to `0`

Comment (pbnj) [reply lua/lint/linters/snyk_iac.lua]: I believe columns start at 1 in (neo)vim.

For example, place the cursor in the 1st column in a buffer and press <kbd>g</kbd> + <kbd>ctrl</kbd><kbd>g</kbd>. Example result: `Col 1 of 105; Line 467 of 48520; Word 1474 of 145763; Byte 49180 of 5216916`

Comment (mfussenegger) [reply lua/lint/linters/snyk_iac.lua]: It uses a mix of indexing. Some is (0,0), some is (1,0). 
Diagnostics are 0 based, from `:help diagnostic-structure`:

> Diagnostics use the same indexing as the rest of the Nvim API (i.e. 0-based
> rows and columns). |api-indexing|

Comment (pbnj) [reply lua/lint/linters/snyk_iac.lua]: I see. Thank you for clarifying this. I will fix this.

**Pull request merged**